glustercli/cmd/tracing.go

```diff
@@ -0,0 +1,183 @@
+	// Trace Update
+	traceUpdateCmd.Flags().StringVar(&flagTraceJaegerEndpoint, "jaeger-endpoint", "", "jaeger endpoint")
+	traceUpdateCmd.Flags().StringVar(&flagTraceJaegerAgentEndpoint, "jaeger-agent-endpoint", "", "jaeger agent endpoint")
+	traceUpdateCmd.Flags().IntVar(&flagTraceJaegerSampler, "jaeger-sampler", 2, "jaeger sampler (1: Always sample, 2: Probabilistic)")
+	traceUpdateCmd.Flags().Float64Var(&flagTraceJaegerSampleFraction, "jaeger-sample-fraction", 0.1, "jaeger sample fraction (min: >0.0, max: <1.0)")
+	traceCmd.AddCommand(traceEnableCmd)
+	traceCmd.AddCommand(traceStatusCmd)
+	traceCmd.AddCommand(traceUpdateCmd)
+	traceCmd.AddCommand(traceDisableCmd)
+}
+
+var traceCmd = &cobra.Command{
+	Use:   "trace",
+	Short: helpTraceCmd,
+}
+
+var traceEnableCmd = &cobra.Command{
+	Use:   "enable <jaeger-endpoint> <jaeger-agent-endpoint>",
+	Short: helpTraceEnableCmd,
+	Args:  cobra.ExactArgs(2),
+	Run: func(cmd *cobra.Command, args []string) {
+		jaegerEndpoint := args[0]
+		jaegerAgentEndpoint := args[1]
+
+		if jaegerEndpoint == "" || jaegerAgentEndpoint == "" {
+			err := errors.New("One or more Jaeger endpoints not specified")
+			failure(errTraceEnableReqFailed, err, 1)
+		}
+
+		if flagTraceJaegerSampler <= 0 || flagTraceJaegerSampler > 2 {
+			err := errors.New("Invalid Jaeger sampler specified")
+			failure(errTraceEnableReqFailed, err, 1)
+		}
+
+		if flagTraceJaegerSampler == 2 {
+			if flagTraceJaegerSampleFraction <= 0.0 || flagTraceJaegerSampleFraction >= 1.0 {
+				err := errors.New("Invalid Jaeger sample fraction specified")
+				failure(errTraceEnableReqFailed, err, 1)
+			}
+		}
+
+		_, err := client.TraceEnable(tracemgmtapi.SetupTracingReq{
+			JaegerEndpoint:       jaegerEndpoint,
+			JaegerAgentEndpoint:  jaegerAgentEndpoint,
+			JaegerSampler:        flagTraceJaegerSampler,
+			JaegerSampleFraction: flagTraceJaegerSampleFraction,
+		})
+
+		if err != nil {
+			if GlobalFlag.Verbose {
+				log.WithError(err).WithFields(log.Fields{
+					"JaegerEndpoint":       jaegerEndpoint,
+					"JaegerAgentEndpoint":  jaegerAgentEndpoint,
+					"JaegerSampler":        flagTraceJaegerSampler,
+					"JaegerSampleFraction": flagTraceJaegerSampleFraction,
+				}).Error("Trace Enable failed")
+			}
+			failure(errTraceEnableReqFailed, err, 1)
+		}
+		fmt.Println("Trace enable successful")
+	},
+}
+
+func samplerTypeToString(sampler int) string {
+	switch sampler {
+	case 0:
+		return "Never"
+	case 1:
+		return "Always"
+	case 2:
+		return "Probabilistic"
+	default:
+		return "Unknown"
+	}
+}
+
+var traceStatusCmd = &cobra.Command{
+	Use:   "status",
+	Short: helpTraceStatusCmd,
+	Run: func(cmd *cobra.Command, args []string) {
+		jaegerConfigInfo, err := client.TraceStatus()
+		if err != nil {
+			failure("Error getting trace status", err, 1)
+		}
+
+		table := tablewriter.NewWriter(os.Stdout)
+		table.SetAlignment(tablewriter.ALIGN_LEFT)
```

Comment:
can you paste the sample output of status?

Reply:
Sample output:
```
$./glustercli trace status --endpoints "http://192.168.122.10:24007" -v
+------------------------+----------------------------+
|      TRACE OPTION      |           VALUE            |
+------------------------+----------------------------+
| Status                 | enabled                    |
| Jaeger Endpoint        | http://192.168.122.1:14268 |
| Jaeger Agent Endpoint  | http://192.168.122.1:6831  |
| Jaeger Sampler         | 1 (Always)                 |
| Jaeger Sample Fraction | 1.00                       |
+------------------------+----------------------------+
```